File: projekt2/src/proj/java/model/Plant.java
package proj.java.model;

import java.util.Random;

public abstract class Plant extends Organism {

    private static final int PLANT_INITIATIVE = 0;

    protected abstract void spread(int newRow, int newCol, World world);

    protected void trySpread(World world) {
        boolean spreadIsValid = false;
        int newRow = 0;
        int newCol = 0;

        Random random = new Random();
        while (spreadIsValid == false) {
            newRow = random.nextInt(world.size());
            newCol = random.nextInt(world.size());
            if (world.getOrganism(newRow, newCol) == null)
                spreadIsValid = true;
        }
        String message = String.format("%s rozprzestrzenil sie na pole (%d , %d)", this.getName(), newCol, newRow);
        world.addMessage(message);
        spread(newRow, newCol, world);
    }

    @Override
    public boolean isAnAnimal() {
        return false;
    }

    @Override
    public int getInitiative() {
        return PLANT_INITIATIVE;
    }

    @Override
    public boolean isHuman() {
        return false;
    }

    @Override
    public void takeAction(World world) {
        var random = new Random();
        if (random.nextInt(100) < 10) {
            trySpread(world);
        }
    }

}

File: projekt3/Plant.py
from random import random, randrange
from Organism import *

class Plant(Organism):
    __CONST_PLANT_INITIATIVWE__ = 0
    
    def __init__(self, row,col, strength, world):
        super().__init__(row,col, strength, self.__CONST_PLANT_INITIATIVWE__, world)
    
    @abstractmethod
    def spread(self, newRow, newCol):
        pass
    
    def trySpread(self):
        spreadIsValid = False
        newRow = 0
        newCol = 0
        while (spreadIsValid == False):
            newRow = randrange(self._world_.size())
            newCol = randrange(self._world_.size())
            if (self._world_.getOrgFromPos(newRow, newCol) == None):
                spreadIsValid = True
        event = self.getName() + " rozprzestrzenil sie na pole" + "(" + str(newRow) + "," + str(newCol) + ")"
        self._world_.addEvent(event)
        self.spread(newRow, newCol)

    def takeAction(self):
        random = randrange(100)
        if (random < 10):
            self.trySpread()
     